torch/random.py

```diff
@@ -101,23 +101,27 @@ def initial_seed() -> int:
 
 
 @contextlib.contextmanager
-def fork_rng(devices=None, enabled=True, _caller="fork_rng", _devices_kw="devices") -> Generator:
+def fork_rng(devices=None, enabled=True, _caller="fork_rng", _devices_kw="devices", device_type="cuda") -> Generator:
     """
     Forks the RNG, so that when you return, the RNG is reset
     to the state that it was previously in.
 
     Args:
-        devices (iterable of CUDA IDs): CUDA devices for which to fork
-            the RNG.  CPU RNG state is always forked.  By default, :meth:`fork_rng` operates
+        devices (iterable of Device IDs): devices for which to fork
+            the RNG. CPU RNG state is always forked. By default, :meth:`fork_rng` operates
             on all devices, but will emit a warning if your machine has a lot
             of devices, since this function will run very slowly in that case.
             If you explicitly specify devices, this warning will be suppressed
         enabled (bool): if ``False``, the RNG is not forked.  This is a convenience
             argument for easily disabling the context manager without having
             to delete it and unindent your Python code under it.
+        deivce(str): device type str, default is `cuda`.
     """
 
-    import torch.cuda
+    device_type = torch.device(device_type).type
+    device_mod = getattr(torch, device_type, None)
+    if device_mod is None:
+        raise RuntimeError("`torch` has no module of `device`, please define torch.{device_type} module")
```

Comment:
A more actionable error would probably include mention that the backend needs to run `torch._register_device_module()` (and link to the pytorch code with the existing docs on open registration)